### PR TITLE
[FUSION-295] Fixes for error handling

### PR DESCRIPTION
**Pull request by akamimoor**
## JIRA ticket
[FUSION-295](https://janrain.atlassian.net/browse/FUSION-295)

## Changes

- Updated the error handling
- Finished up integration for the Login

Update AppAuth-iOS SDK:
`carthage update --use-xcframeworks --platform iOS`

Confluence doc for development instructions: [Here](https://janrain.atlassian.net/l/cp/wqc2506v)

[FUSION-295]: https://janrain.atlassian.net/browse/FUSION-295?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: Examples/Example-iOS_ObjC-Carthage/Source/Authentication/Authenticator.m [hunk 1]
```diff
@@ -227,13 +226,13 @@ - (void)finishLoginWithAuthStateResponse:(nullable OIDAuthState *)authState comp
     } else {
         [self.authStateManager setAuthState:nil];
         [self.authStateManager setBrowserState:StateInactive];
-        completion([[AuthError alloc] initWithType:AuthErrorTypeNoAuthState]);
+        completion([[AuthError alloc] initWithType:AuthErrorTypeUserCancelledAuthorizationFlow]);
```
Comment (jbush-janrain) [inline Examples/Example-iOS_ObjC-Carthage/Source/Authentication/Authenticator.m:229]: It's the smallest of requests, but `Cancelled` should be switched to `Canceled`.

File: Examples/Example-iOS_ObjC-Carthage/Source/Utitlities/AuthError.m [hunk 2]
```diff
@@ -59,72 +59,72 @@ - (NSString *)localizedDescription {
...
-            return NSLocalizedString(@"Invalid HTTP response object received after an API call.", nil);
+            return NSLocalizedString(@"Invalid HTTP response object received after an API call", nil);
         }
         case AuthErrorTypeMissingIdToken: {
-            return NSLocalizedString(@"ID token needed to fulfill this operation.", nil);
+            return NSLocalizedString(@"ID token needed to fulfill this operation", nil);
         }
         case AuthErrorTypeUserCancelledAuthorizationFlow: {
-            return NSLocalizedString(@"The redirect request was cancelled.", nil);
+            return NSLocalizedString(@"The redirect request was cancelled by the user", nil);
```
Comment (jbush-janrain) [inline Examples/Example-iOS_ObjC-Carthage/Source/Utitlities/AuthError.m:115]: Ditto here.  Now realizing this is how the code was, so no sweat if it's left. I think the meaning is easily understood either way.

**Review by jbush-janrain — APPROVED** [review]
✅ Code changes look reasonable, left a very small comment on potentially updating the existing spelling but nothing game changing. 
✅ Tests are building and passing locally on XCode:
test_passing_in_xcode

File: Examples/Example-iOS_ObjC-Carthage/Source/Authentication/Authenticator.m (hunk 1 above)
Comment (schaeferCode) [reply Examples/Example-iOS_ObjC-Carthage/Source/Authentication/Authenticator.m:229]: Never knew: https://www.merriam-webster.com/grammar/canceled-or-cancelled

**Review by schaeferCode — APPROVED** [review]
Looks good!

**Review by kreese-akamai — APPROVED** [review]
⚡

**Comment by kreese-akamai** [comment]
"canceled" or "cancelled" ... Both spellings are correct!
- Americans favor canceled _(single "l")_
- British English and other dialects favour cancelled _(double "l")_
https://www.merriam-webster.com/grammar/canceled-or-cancelled

File: Examples/Example-iOS_ObjC-Carthage/Source/Authentication/Authenticator.m (hunk 1 above)
Comment (kreese-akamai) [reply Examples/Example-iOS_ObjC-Carthage/Source/Authentication/Authenticator.m:229]: Indeed, both spellings are correct!
- Americans favor canceled (single "l")
- British English and other dialects favour cancelled (double "l")

Comment (jbush-janrain) [reply Examples/Example-iOS_ObjC-Carthage/Source/Authentication/Authenticator.m:229]: I had no idea either!  Great info!

**Pull request merged**